chapters/scoping.tex

```diff
@@ -98,7 +98,13 @@ \subsection{Composite Name Lookup}\label{composite-name-lookup}
 \item
   If the identifier denotes a class, that class is temporarily flattened (as if instantiating a component without modifiers of this class, see \cref{modification-environment}) and using the enclosing classes of the denoted class.
   The rest of the name (e.g., \lstinline!B! or \lstinline!B.C!) is looked up among the declared named elements of the temporary flattened class.
-  If the class does not satisfy the requirements for a package, the lookup is restricted to encapsulated\indexinline{encapsulated} elements only.
+  The restrictions for the lookup are:
+  \begin{itemize}
+  \item If the class is declared as \lstinline!package! or \lstinline!operator! (see below) all elements can be found.
+  \item If the element is declared as \lstinline!encapsulated!\indexinline{encapsulated} it can also be found in other cases.
+  \item A deprecated case is that if the class satisfies the requirements for a \lstinline!package! (without being declared as such), it is still treated as a \lstinline!package!.
+  \item Otherwise it is an error.
+  \end{itemize}
```

Comment:
The current formulation about encapsulated sounds as if the rules for encapsulated elements are given by cases described somewhere else.  Also, it's odd that the restrictions come down to listing what is not restricted.
```suggestion
  The lookup may succeed only in the following cases:
  \begin{itemize}
  \item An element can be found if it is declared as \lstinline!encapsulated!\indexinline{encapsulated}.
  \item If the class is declared as \lstinline!package! or \lstinline!operator! (see below) all elements can be found.
  \item A deprecated case is that if the class satisfies the requirements for a \lstinline!package! (without being declared as such), it is still treated as a \lstinline!package!.
  \end{itemize}
```

Reply:
This seems like a too major change and there are at least two problems with it:

- Lists are often ordered in terms of relevance, and thus leading with the package-case is preferable. The 'also be found in other cases' can be skipped with re-ordering the alternatives.
- 'may succeed only' seems like an odd construct. 'The lookup will only find the element in the following cases: (with slight rewording)

Reply:
Resolving problems as suggested:
```suggestion
  The lookup will only find the element in the following cases:
  \begin{itemize}
  \item If the class is declared as \lstinline!package! or \lstinline!operator! (see below) all elements can be found.
  \item An element can be found if it is declared as \lstinline!encapsulated!\indexinline{encapsulated}.
  \item A deprecated case is that if the class satisfies the requirements for a \lstinline!package! (without being declared as such), it is still treated as a \lstinline!package!.
  \end{itemize}
```

Reply:
Regarding "may succeed", I didn't think we wanted to give the impression that the element necessarily exists.  Now, _the element_ sounds as we take its existence for granted, but maybe that's not a big deal as it is obvious that it may not exist?